src/compiler/parser.ts

```diff
@@ -5698,7 +5699,7 @@ namespace Parser {
      *      1) UnaryExpression[?Yield]
      *      2) UpdateExpression[?Yield] ** ExponentiationExpression[?Yield]
      */
-    function parseUnaryExpressionOrHigher(): UnaryExpression | BinaryExpression {
+    function parseUnaryExpressionOrHigher(previousNode?: Node): UnaryExpression | BinaryExpression {

```

Comment:
It wouldn't be a 100% straightforward "replace the `?` with `| undefined` then add `/*previousNode*/ undefined` to function calls because of a `nextTokenAnd(parseLeftHandSideExpressionOrHigher)`. But it's 13 one-line changes otherwise: https://github.com/JoshuaKGoldberg/TypeScript/commit/2f6b35e5b9bca1c887d6b0a54d7558dc1a6b9dfc

Reply:
Thanks. The main thing this points out to me is that there is only one narrow code path that preserves the previous node, and a bunch of others that could, but don't. I guess nothing else needs this information, but it still seems wasteful to wire it up for one usage.